presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -472,11 +468,27 @@ private static int findGroupEnd(PagesIndex pagesIndex, PagesHashStrategy pagesHa
             return pagesIndex.getPositionCount();
         }
 
+        return findEndPosition(startPosition, pagesIndex.getPositionCount(), (firstPosition, secondPosition) -> pagesIndex.positionEqualsPosition(pagesHashStrategy, firstPosition, secondPosition));
+    }
+
+    /**
+     * @param startPosition - inclusive
+     * @param endPosition - exclusive
+     * @param comparator - returns true if positions given as parameters are equal
+     * @return the end of the group position exclusive (the position the very next group starts)
+     */
+    @VisibleForTesting
+    static int findEndPosition(int startPosition, int endPosition, BiFunction<Integer, Integer, Boolean> comparator)
```

Comment:
`BiFunction<T, U, Boolean>` is more natural as `BiPredicate<T, U>` and avoids boxing